.github/workflows/maturin_release.yaml

```diff
@@ -0,0 +1,128 @@
+name: CI
```

Comment:
This file was initially produced using the [maturin generate-ci command,](https://www.maturin.rs/distribution.html#github-actions) but I made some manual edits to try and match the versions and architectures we use to build our python client lib: https://github.com/sift-stack/sift/blob/main/.github/workflows/python_build.yaml#L44-L48